src/get-polyfill.ts

```diff
@@ -0,0 +1,666 @@
+        entry,
+        IS_PERSIST
+    );
+  }
+
+  const navigation: Navigation = givenNavigation ?? new NavigationPolyfill({
+    entries: initialEntries,
+    currentIndex: initialMeta?.currentIndex,
+    currentKey: initialMeta?.key,
+    getState,
+    setState,
+    disposeState
+  });
+
+  const pushState = history?.pushState.bind(history);
+  const replaceState = history?.replaceState.bind(history);
+  const historyGo = history?.go.bind(history);
+  // const back = history?.back.bind(history);
+  // const forward = history?.forward.bind(history);
+
+  // const origin = typeof location === "undefined" ? "https://example.com" : location.origin;
+
+  return {
+    navigation,
+    history,
+    async apply() {
+      console.log("APPLYING POLYFILL TO NAVIGATION");
+
+      if (
+          isNavigationPolyfill(givenNavigation) &&
+          IS_PERSIST &&
+          historyInitialState &&
+          initialMeta
+      ) {
+        givenNavigation[NavigationSetEntries](initialEntries);
+        givenNavigation[NavigationSetCurrentKey](initialMeta.key);
+      } else if (!navigation.length && isNavigationPolyfill(navigation)) {
+        // Initialise empty navigation
+        navigation[NavigationSetEntries]([
+          {
+            key: v4()
+          }
+        ]);
+      }
+
+      if (HISTORY_INTEGRATION) {
+        const ignorePopState = new Set<string>();
+        const ignoreCurrentEntryChange = new Set<string>();
+
+        navigation.addEventListener("currententrychange", ({ navigationType, from }) => {
+          const { currentEntry } = navigation;
+          if (!currentEntry) return;
+          const { key, url } = currentEntry;
+          if (ignoreCurrentEntryChange.delete(key) || !currentEntry?.sameDocument) return;
+
+          const historyState = getState(currentEntry);
+
+          switch (navigationType) {
+            case "push":
+              return pushState(historyState, "", url)
+            case "replace":
+              return replaceState(historyState, "", url)
+            case "traverse":
+              const delta = currentEntry.index - from.index;
+              ignorePopState.add(key);
+              return historyGo(delta);
+            case "reload":
+              // TODO
+          }
+        });
+
+        window.addEventListener("popstate", (event) => {
+          const { state, originalState } = event;
+          const foundState = originalState ?? state;
+          if (!isStateHistoryWithMeta(foundState)) return;
+          const {
+            [NavigationKey]: {
+              key
+            }
+          } = foundState
+          if (ignorePopState.delete(key)) return;
+          ignoreCurrentEntryChange.add(key);
+          let committed;
+          try {
+            committed = navigation.traverseTo(key).committed;
+          } catch (error) {
+            if (error instanceof InvalidStateError && !PERSIST_ENTRIES) {
+              // ignore the error
+              return;
+            }
```

Comment:
Btw, the the reason for ignoring this error is that this is common when not persisting entries.